src/SignalR/clients/csharp/Client.Core/src/IHubConnection.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    /// <summary>
+    /// A connection used to invoke hub methods on a SignalR Server.
+    /// </summary>
+    /// <remarks>
+    /// A <see cref="HubConnection"/> should be created using <see cref="HubConnectionBuilder"/>.
+    /// </remarks>
+    public interface IHubConnection
```

Comment:
I have only extracted methods needed for purposes of this PR into `IHubConnection`. Let me know if there are other methods you would like to have extracted as part of this. (e.g. `HubConnection.StartAsync` ?)